### ISSUE TITLE
M2M field has display issues with nested M2O field (which has a manual primary key)

### ISSUE THREAD

**Issue by HitomiTenshi** [directus/directus#10647]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [ ] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.
- [X] Theres no issue that described it as detailed as I did 😜 

### Describe the Bug

I have 3 collections: `accounts` -> M2M `addresses` -> M2O `countries` (countries has a manual primary key).

While editing an `account`: When creating a new address, or when selecting an existing address, the display configuration is showing `--` for the nested M2O field.

If there was a previously selected `address` and you remove and re-select the same `address` again, this issue does not occur. It only occurs if you select a different `address` or create a new one. The display bug will always occur when creating a new account, because initially there is no address selected.

Related to issue #8139

There were multiple bugs going on in that issue so I wanted to create a new issue that explains the display bug more concisely.

Edit: I just saw that it is also related to: #10476

### To Reproduce

Create the following collections:

1. countries (with manual primary key called `code`, rest at default values)
2. addresses (default values)
3. accounts (default values)

Setup the fields:

- In `countries` add **string** field `country`
- In `addresses` add **M2O** field `country` -> related collection is `countries`
- - Set relational triggers to delete (cascade)
- - Set Display and Interface to show the following template: "`code`: `country`"
- In `addresses` add **string** field `address`
- In `accounts` add **M2M** field `addresses` -> related collection is `addresses`
- - Set all relational triggers to delete (cascade)
- - Set Display and Interface to show the following template: "`addresses_id.country.code`: `addresses_id.address`"

Now add countries, addresses and at the end an account which references to an address. You will see that when selecting an address the country code is being displayed as `--` initially. Only after saving will it display normally again.

If you edit the account you just created and de-select and re-select the same address this issue will not occur.

See video (if you want to skip collection setup go to 02:14):

https://user-images.githubusercontent.com/7304212/147173918-eeb83531-b3fa-4ea5-9999-70d98a1e5ca2.mp4

### Errors Shown

N/A

### What version of Directus are you using?

9.3.0

### What version of Node.js are you using?

16.13.1

### What database are you using?

MariaDB 10.3.32

### What browser are you using?

Firefox Latest

### What operating system are you using?

Windows 10

### How are you deploying Directus?

Ubuntu Server 20.04, npm installation with a systemd service running it

**Comment by shankiflang** [comment]
Hi, I'm here to see if there are any updates on the subject? I see that the post is from 2021 and it's still not resolved :) Just to know where the topic is at!

**Comment by rijkvanzanten** [comment]
[**Unfortunately, we can not provide estimates on features or fixes**](https://docs.directus.io/getting-started/support.html#can-you-give-an-eta-for-this-feature-fix). As of now, the only way to get an accurate timeline or release date is to [sponsor](https://docs.directus.io/getting-started/support.html#sponsored-work) this task. We recognize that this may be an important feature/fix, but we are a small open-source organization with a lot to triage and complete. If you can't sponsor this ticket, then the next best thing is to increase its priority by giving it a 👍 . Or contribute a pull-request... after all, this is OSS!

---

_This is an automated response._

**Comment by rijkvanzanten** [comment]
Linear: [ENG-282](https://linear.app/directus/issue/ENG-282/m2m-field-has-display-issues-with-nested-m2o-field-which-has-a-manual)
